src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1405,18 +1405,45 @@ abstract class RefChecks extends Transform {
       }
     }
 
-    private def checkAnnotations(tpes: List[Type], tree: Tree) = tpes foreach { tp =>
-      checkTypeRef(tp, tree, skipBounds = false)
-      checkTypeRefBounds(tp, tree)
-    }
-    private def doTypeTraversal(tree: Tree)(f: Type => Unit) = if (!inPattern) tree.tpe foreach f
+    private def applyRefchecksToAnnotations(tree: Tree): Tree = {
+      def applyChecks(annots: List[AnnotationInfo]): List[AnnotationInfo] = {
+        annots.foreach { ann =>
+          checkTypeRef(ann.tpe, tree, skipBounds = false)
+          checkTypeRefBounds(ann.tpe, tree)
+        }
 
-    private def applyRefchecksToAnnotations(tree: Tree): Unit = {
-      def applyChecks(annots: List[AnnotationInfo]) = {
-        checkAnnotations(annots map (_.atp), tree)
-        transformTrees(annots flatMap (_.args))
+        annots
+          .map(_.transformArgs(transformTrees))
+          .groupBy(_.symbol)
+          .flatMap((groupRepeatableAnnotations _).tupled)
+          .toList
       }
 
+      def groupRepeatableAnnotations(sym: Symbol, anns: List[AnnotationInfo]): List[AnnotationInfo] =
+        if (!(sym isSubClass ClassfileAnnotationClass)) anns else anns match {
+          case single :: Nil => anns
+          case multiple      => // groupBy won't give us `Nil`s
```

Comment:
stale comment

Reply:
Not stale, just out of place, like a slice of Limburger on a McDonald's happy meal patty. It was meant to refer to this method being used in `groupBy` a few lines up.

Reply:
Ah sorry, i searched for `groupBy` in the diff, but missed it.